build/build/src/ide/web/fonts.rs

```diff
@@ -0,0 +1,89 @@
+use crate::prelude::*;
+
+use crate::ide::web::download::get_file_from_cache_or_download;
+use crate::ide::web::google_font;
+
+use enso_enso_font::ttf;
+use enso_font::NonVariableFaceHeader;
+use ide_ci::cache::Cache;
+
+
+
+// =========================
+// === HTML Font Support ===
+// =========================
+
+pub async fn install_html_fonts(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    output_path: impl AsRef<Path>,
+) -> Result {
+    let output_path = output_path.as_ref();
+    google_font::download_google_font(cache, octocrab, "mplus1", output_path).await?;
+    install_enso_font_for_html(cache, octocrab, output_path).await?;
+    Ok(())
+}
+
+pub async fn install_enso_font_for_html(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    output_path: impl AsRef<Path>,
+) -> Result {
+    let output_path = output_path.as_ref();
+    let html_fonts: HashMap<_, _> = [
+        (NonVariableFaceHeader { weight: ttf::Weight::Normal, ..default() }, "Regular"),
+        (NonVariableFaceHeader { weight: ttf::Weight::ExtraBold, ..default() }, "Bold"),
+    ]
+    .into_iter()
+    .collect();
+    let html_font_definitions = enso_enso_font::enso_font()
+        .variations()
+        .filter(|v| html_fonts.contains_key(&v.header))
+        .collect();
+    let get_font_files = async {
+        let archive = get_enso_font_package_(cache, octocrab.clone()).await?;
+        enso_enso_font::extract_fonts(&html_font_definitions, archive, output_path).await
+    };
+    let make_css_file = async {
+        let mut css = String::new();
+        let family = "Enso";
+        let url = ".";
+        for (header, variant) in html_fonts {
+            use std::fmt::Write;
+            let def = html_font_definitions.get(header);
+            let def = def.ok_or(anyhow!("Required font not found in package."))?;
```

Comment:
Use `with_context`.
Also, it'd nice to say what font is missing from which package.